src/downloader/YoutubeDownloader.py

```diff
@@ -19,6 +20,12 @@ def __init__(self, parents=None, *args, dl_archive=os.path.join(os.getcwd(), 'yd
     def download(self, md_item: MetaDataItem) -> VideoItem:
         ydl = youtube_dl.YoutubeDL(self.dl_opts)
 
+        new_name = self.video_storage.get_file_name(md_item)
```

Comment:
could we put this inside the universal downloader instead? This way none of the downloaders would run if the video is already downloaded

Reply:
Turns out there's already something to handle this in the universal downloader but the logic was broken because we didn't consider file extensions. Fixed now.